test/no-useless-switch-case.mjs

```diff
@@ -0,0 +1,179 @@
+import outdent from 'outdent';
+import {getTester} from './utils/test.mjs';
+
+const {test} = getTester(import.meta);
+
+test.snapshot({
+	valid: [
+		outdent`
+			switch (foo) {
+				case a:
+				case b:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a:
+					handleCaseA();
+					break;
+				default:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a:
+					handleCaseA();
+				default:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a:
+					break;
+				default:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a:
+					handleCaseA();
+					// Fallthrough
+				default:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a:
+				default:
+					handleDefaultCase();
+					break;
+				case b:
+					handleCaseB();
+					break;
+			}
+		`,
+		outdent`
+			switch (1) {
+					// This is not useless
+					case 1:
+					default:
+							console.log('1')
+					case 1:
+							console.log('2')
+			}
+		`,
+	],
+	invalid: [
+		outdent`
+			switch (foo) {
+				case a:
+				default:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a: {
+				}
+				default:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a: {
+					;;
+					{
+						;;
+						{
+							;;
+						}
+					}
+				}
+				default:
+					handleDefaultCase();
+					break;
+			}
+		`,
+		outdent`
+			switch (foo) {
+				case a:
+				case (( b ))         :
```

Comment:
```suggestion
				case (( b )):
```
Curious were those spaces intentional?

Reply:
Yes

Reply:
For this function https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1779/files#diff-d14f0af39629703585e6c7779ca7b3c2a8207c63c330bc60f2849cb40a8e14f2R16, so people can't refactor to use something like `end + 1`